pkg/analyze/velero.go

```diff
@@ -0,0 +1,648 @@
+				}
+				result.IsFail = true
+				result.Message = fmt.Sprintf("Volume Snapshot Location %s phase is %s", volumeSnapshotLocation.Name, volumeSnapshotLocation.Status.Phase)
+				results = append(results, result)
+				failures++
+			}
+		}
+		if failures == 0 {
+			results = append(results, &AnalyzeResult{
+				Title:   "Velero Volume Snapshot Locations",
+				IsPass:  true,
+				Message: fmt.Sprintf("Found %d volume snapshot locations", len(volumeSnapshotLocations)),
+			})
+		}
+	}
+
+	return results
+}
+
+func analyzeLogs(logs map[string][]byte) []*AnalyzeResult {
+	results := []*AnalyzeResult{}
+	if len(logs) > 0 {
+		for _, logBytes := range logs {
+			logContent := string(logBytes)
+			result := &AnalyzeResult{
+				Title: fmt.Sprintf("Velero logs for pod [node-agent]"),
+			}
+			if strings.Contains(logContent, "permission denied") {
+				result.IsWarn = true
+				result.Message = fmt.Sprintf("Found 'permission denied' in node-agent* pod log file(s)")
+				results = append(results, result)
+				continue
+			}
+
+			if strings.Contains(logContent, "error") || strings.Contains(logContent, "panic") || strings.Contains(logContent, "fatal") {
+				result.IsWarn = true
+				result.Message = fmt.Sprintf("Found error|panic|fatal in node-agent* pod log file(s)")
+				results = append(results, result)
+			}
+		}
+
+		results = append(results, &AnalyzeResult{
+			Title:   "Velero Logs analysis",
+			IsPass:  true,
+			Message: fmt.Sprintf("Found %d log files", len(logs)),
+		})
+	}
+	return results
+}
+
+func aggregateResults(results []*AnalyzeResult) []*AnalyzeResult {
+	out := []*AnalyzeResult{}
+	resultFailed := false
+	for _, result := range results {
+		if result.IsFail {
+			resultFailed = true
+		}
+		out = append(out, result)
+	}
+	if len(results) > 0 {
+		if resultFailed == false {
+			out = append(out, &AnalyzeResult{
+				Title:   "Velero Status",
+				IsPass:  true,
+				Message: "Velero setup is healthy",
+			})
+		}
+		if resultFailed == true {
+			out = append(out, &AnalyzeResult{
+				Title:   "Velero Status",
+				IsWarn:  true,
+				Message: "Velero setup is not entirely healthy",
+			})
+		}
+	}
+
+	return out
+}
+
+func GetVeleroBackupsDirectory() string {
+	return "cluster-resources/custom-resources/backups.velero.io"
+}
+
+func GetVeleroBackupRepositoriesDirectory() string {
+	return "cluster-resources/custom-resources/backuprepositories.velero.io"
+}
+
+func GetVeleroBackupStorageLocationsDirectory() string {
+	return "cluster-resources/custom-resources/backupstoragelocations.velero.io"
+}
+
+func GetVeleroDeleteBackupRequestsDirectory() string {
+	return "cluster-resources/custom-resources/deletebackuprequests.velero.io"
+}
+
+func GetVeleroDownloadRequestsDirectory() string {
+	return "cluster-resources/custom-resources/downloadrequests.velero.io"
+}
+
+func GetVeleroLogsDirectory(namespace string) string {
+	return fmt.Sprint("%s/logs", namespace)
```

Comment:
typo error, should be `fmt.Sprintf`, right?

Reply:
Ah yea I pushed it while making another change. But thanks for the find! 

Removing namespace entirely from here because no use. 

I'll push a change soon and it'll be clear.

Reply:
This was also caught in the check schema test!